graylog2-server/src/main/java/org/graylog2/rest/resources/system/ClusterConfigResource.java

```diff
@@ -117,27 +125,44 @@ public Response update(@ApiParam(name = "configClass", value = "The name of the
                            @NotNull InputStream body) throws IOException {
         final Class<?> cls = classFromName(configClass);
         if (cls == null) {
-            throw new NotFoundException("Couldn't find configuration class \"" + configClass + "\"");
+            throw new NotFoundException(createNoClassMsg(configClass));
         }
 
-        final Object o;
+        final Object configObject = parseConfigObject(configClass, body, cls);
+        validateConfigObject(configObject);
+        writeConfigObject(configClass, configObject);
+
+        return Response.accepted(configObject).build();
+    }
+
+    private void writeConfigObject(String configClass, Object configObject) {
         try {
-            o = objectMapper.readValue(body, cls);
+            clusterConfigService.write(configObject);
         } catch (Exception e) {
-            final String msg = "Couldn't parse cluster configuration \"" + configClass + "\".";
+            final String msg = "Couldn't write cluster config \"" + configClass + "\".";
```

Comment:
Great idea to use a string formatter in the other messages in this class. Can a string formatter be used here too?